--- crypto/engine/eng_list.orig.c	2019-05-28 21:12:21.000000000 +0800
+++ crypto/engine/eng_list.c	2019-07-15 13:51:33.949647300 +0800
@@ -318,7 +318,11 @@
      */
     if (strcmp(id, "dynamic")) {
         if ((load_dir = ossl_safe_getenv("OPENSSL_ENGINES")) == NULL)
+        {
+#ifdef ENGINESDIR
             load_dir = ENGINESDIR;
+#endif
+        }
         iterator = ENGINE_by_id("dynamic");
         if (!iterator || !ENGINE_ctrl_cmd_string(iterator, "ID", id, 0) ||
             !ENGINE_ctrl_cmd_string(iterator, "DIR_LOAD", "2", 0) ||
